libs/javalib/worker/src/mill/javalib/zinc/ZincWorkerRpcServer.scala

```diff
@@ -0,0 +1,224 @@
+          case msg: ClientToServer.CompileMixed =>
+            compileMixed(requestId, msg).asInstanceOf[input.Response]
+          case msg: ClientToServer.DocJar => docJar(requestId, msg).asInstanceOf[input.Response]
+        }
+      }
+
+      private def compileJava(
+          clientRequestId: MillRpcRequestId,
+          msg: ClientToServer.CompileJava
+      ): msg.Response = {
+        worker.compileJava(
+          upstreamCompileOutput = msg.upstreamCompileOutput,
+          sources = msg.sources,
+          compileClasspath = msg.compileClasspath,
+          javacOptions = msg.javacOptions,
+          reporter = reporterAsOption(clientRequestId, msg.reporterMode),
+          reportCachedProblems = msg.reporterMode.reportCachedProblems,
+          incrementalCompilation = msg.incrementalCompilation
+        )(using msg.ctx, deps)
+      }
+
+      private def compileMixed(
+          clientRequestId: MillRpcRequestId,
+          msg: ClientToServer.CompileMixed
+      ): msg.Response = {
+        worker.compileMixed(
+          upstreamCompileOutput = msg.upstreamCompileOutput,
+          sources = msg.sources,
+          compileClasspath = msg.compileClasspath,
+          javacOptions = msg.javacOptions,
+          scalaVersion = msg.scalaVersion,
+          scalaOrganization = msg.scalaOrganization,
+          scalacOptions = msg.scalacOptions,
+          compilerClasspath = msg.compilerClasspath,
+          scalacPluginClasspath = msg.scalacPluginClasspath,
+          reporter = reporterAsOption(clientRequestId, msg.reporterMode),
+          reportCachedProblems = msg.reporterMode.reportCachedProblems,
+          incrementalCompilation = msg.incrementalCompilation,
+          auxiliaryClassFileExtensions = msg.auxiliaryClassFileExtensions,
+          compilerBridgeData = clientRequestId
+        )(using msg.ctx, deps)
+      }
+
+      private def docJar(
+          clientRequestId: MillRpcRequestId,
+          msg: ClientToServer.DocJar
+      ): msg.Response =
+        worker.docJar(
+          scalaVersion = msg.scalaVersion,
+          scalaOrganization = msg.scalaOrganization,
+          compilerClasspath = msg.compilerClasspath,
+          scalacPluginClasspath = msg.scalacPluginClasspath,
+          args = msg.args,
+          compilerBridgeData = clientRequestId
+        )
+    }
+  }
+}
+object ZincWorkerRpcServer {
+
+  /**
+   * @param taskDest the task's destination folder.
+   */
+  case class Initialize(taskDest: os.Path, jobs: Int, compileToJar: Boolean, zincLogDebug: Boolean)
+      derives ReadWriter
+
+  sealed trait ReporterMode derives ReadWriter {
+    def reportCachedProblems: Boolean
+  }
+  object ReporterMode {
+    case object NoReporter extends ReporterMode {
+      override def reportCachedProblems: Boolean = false
+    }
+
+    case class Reporter(reportCachedProblems: Boolean, maxErrors: Int) extends ReporterMode
+  }
+
+  sealed trait ClientToServer extends MillRpcMessage derives ReadWriter
```

Comment:
Now that we're on Scala 3, can these be `enum`s?

Reply:
Not sure if enums can take `type` members, but if not I think they should be replaceable with type parameters